src/custom-element.js

```diff
@@ -656,6 +657,28 @@ export function createAmpElementProto(win, name, implementationClass) {
     return this.resources_.getResourceForElement(this).getLayoutBox();
   };
 
+ /**
+  * @return {!LayoutRect}
+  */
+  ElementProto.getInsersectionElementLayoutBox = function() {
```

Comment:
Basically I'd like to avoid too much back and forth between custom element and base element classes.